### PR TITLE
Separation of JS and PHP

### PR DESCRIPTION
**Pull request by nikolina-p**
Should I do all "else code" this way? @kunicmarko20

### CODE REVIEW

File: Controller/CategoriesFrontEndController.php [hunk 1]
```diff
@@ -8,11 +8,10 @@ class CategoriesFrontEndController
     public static function insertCategory()
     {
         $naziv = $opis = $nadKat = "";
-        if (!empty($_POST['categoryName_new'])) {
+        if (empty($_POST['categoryName_new'])) {
             $n = CategoriesFrontEndController::test_input_KAT($_POST['categoryName_new']);
         } else {
-            echo "<script>document.getElementById('errName_new').innerHTML = 'Insert name of categroy';
-			document.getElementById('newCategory').style.display = 'inline';</script>";
+            echo self::renderError("../templates/categories_template.php", array("errorMessage" => "Invalid request: Name field is missing."));
```
Comment (kunicmarko20) [inline Controller/CategoriesFrontEndController.php]: `array => []`

File: Controller/CategoriesFrontEndController.php [hunk 2]
```diff
@@ -148,4 +147,25 @@ public static function test_input($data)
         $data = htmlspecialchars($data);
         return $data;
     }
+    
+    public static function renderError(string $templatePath, array $data)
+    {
+        if(file_exists($templatePath)){
```
Comment (kunicmarko20) [inline Controller/CategoriesFrontEndController.php]:
```
if (!file_exists($templatePath)) {
    throw new \Exception();
}
```
Looks better like this, don't you think?

File: Controller/CategoriesFrontEndController.php [hunk 3]
```diff
@@ -148,4 +147,25 @@ public static function test_input($data)
...
+            //Extracts vars to current view scope
+            extract($data);
+
```
Comment (kunicmarko20) [inline Controller/CategoriesFrontEndController.php]: I would rather extract this logic into a class that would be for rendering templates

File: Controller/CategoriesFrontEndController.php [hunk 4]
```diff
@@ -148,4 +147,25 @@ public static function test_input($data)
...
+            //Includes contents
+            include $templatePath;
+            $buffer = ob_get_contents();
+            @ob_end_clean();
+
+            //Returns output buffer
+            return $buffer;
+        } else {
+            //Throws exception
```
Comment (kunicmarko20) [inline Controller/CategoriesFrontEndController.php]: don't leave a comment, do it.

File: Controller/UsersFrontEndController.php [hunk 5]
```diff
@@ -116,7 +116,7 @@ public static function createNewUser()
             Photo::photoUpload("profilePhoto_new", "../public/images/", $user->getId(), $msg, "single");
             echo $msg;
         } else {
-            echo "<script>alert('Greska kod unosa korisnicke slike. ".$msg."');</script>";
+            echo "<script>alert('Greska kod unosa korisnicke slike. ".$msg."');</script>"; //srb
```
Comment (kunicmarko20) [inline Controller/UsersFrontEndController.php]: don't leave a comment, do it.

File: Controller/BaseController.php [hunk 6]
```diff
@@ -0,0 +1,35 @@
+<?php
+namespace Myalpinerocks;
+
+abstract class BaseController
+{
+    public static function renderTemplate(string $templatePath, array $data = [])
```
Comment (kunicmarko20) [inline Controller/BaseController.php:6]: why static?

File: Controller/BaseController.php [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+        //Includes contents
+        include $templatePath;
+        $buffer = ob_get_contents();
+        @ob_end_clean();
+
+        //Returns output buffer
+        return $buffer;            
+    }
+    
+    public static function test_input($data)
```
Comment (kunicmarko20) [inline Controller/BaseController.php:26]: this is more `sanitize_input`

File: Controller/BaseController.php (hunk 6 above)
Comment (nikolina-p) [reply Controller/BaseController.php:6]: My idea was to make all Controller classes static cos they do not have state (they do not keep information about entities like Entity classes do). Controller classes just do things as requested by user, like one-time jobs.

Comment (kunicmarko20) [reply Controller/BaseController.php:6]: `cos they do not have state` that doesn't mean they have to be static

**Pull request merged**